packages/story-editor/src/components/canvas/useInsertElement.js

```diff
@@ -71,7 +71,8 @@ function useInsertElement() {
       }
 
       try {
-        const imageSrcProxied = getProxiedUrl(resource, imageSrc);
+        const needsProxy = await checkResourceAccess(imageSrc);
```

Comment:
Poster images are never hotlinked, so they don't need that. We can really avoid that HTTP request.

Reply:
```suggestion
        const needsProxy = type === 'image' && await checkResourceAccess(imageSrc);
```

a simple fix, albeit not  the best

Reply:
No this is the wrong way around. We want to check for poster images here not images. A poster image on an hot linked image could is a local image ( not needing a proxy ) but a poster image on a coverr or tenor resource may need a proxy.

Reply:
3P media never needs a proxy

Reply:
Okay, but we dont have a simple check for is hotlinked.

Reply:
Sounds like adding `isHotlinked` to resources will make our lives a lot easier

Reply:
> Sounds like adding isHotlinked to resources will make our lives a lot easier 

Let's no add another property like this. We already `local` and `isExternal` which are boolean. We should just add `source`. Values would be `local`, `library`, `hotlink`, `unsplash`, `tenor` or `coverr`. We could then even get rid of  `local` and `isExternal`. We can even add a little helper function called, isExternalResource

Reply:
Interesting suggestion. Something I'd be willing to explore in a dedicated ticket.